rxandroidble/src/test/groovy/com/polidea/rxandroidble/internal/connection/RxBleConnectionTest.groovy

```diff
@@ -655,6 +657,83 @@ class RxBleConnectionTest extends GradleRoboSpecification {
         setupIndicationUuidClosure             | setupNotificationCharacteristicClosure
     }
 
+    def "should pass items emitted by observable returned from RxBleRadioOperationCustom.asObservable()"() {
+        given:
+        def operationBuilder = createCustomOperationBuilder(Observable.just(true, false, true))
+
+        when:
+        objectUnderTest.queue(operationBuilder).subscribe(testSubscriber)
+
+        then:
+        testSubscriber.assertCompleted()
+        testSubscriber.assertValues(true, false, true)
+    }
+
+    def "should release the radio if custom operation will throw out of RxBleRadioOperationCustom.asObservable()"() {
+        given:
+        def operationBuilder = new RxBleRadioOperationCustom<Boolean>() {
+            @Override
+            Observable<Boolean> asObservable(BluetoothGatt bluetoothGatt,
+                                             RxBleGattCallback rxBleGattCallback,
+                                             Scheduler scheduler) throws Throwable {
+                throw new RuntimeException()
+            }
+        }
+
+        when:
+        objectUnderTest.queue(operationBuilder).subscribe(testSubscriber)
+
+        then:
+        testSubscriber.assertError(RuntimeException.class)
+        flatRadio.semaphore.isReleased() >> true
+    }
+
+    def "should release the radio if observable returned from RxBleRadioOperationCustom.asObservable() will emit error"() {
+        given:
+        def operationBuilder = createCustomOperationBuilder(Observable.error(new RuntimeException()))
+
+        when:
+        objectUnderTest.queue(operationBuilder).subscribe(testSubscriber)
+
+        then:
+        testSubscriber.assertError(RuntimeException.class)
+        flatRadio.semaphore.isReleased() >> true
+    }
+
+    def "should release the radio when observable returned from RxBleRadioOperationCustom.asObservable() will complete"() {
+        given:
+        def operationBuilder = createCustomOperationBuilder(Observable.empty())
+
+        when:
+        objectUnderTest.queue(operationBuilder).subscribe(testSubscriber)
+
+        then:
+        testSubscriber.assertCompleted()
+        flatRadio.semaphore.isReleased() >> true
+    }
+
+    def "should pass error if observable returned from RxBleRadioOperationCustom.asObservable() will emit error"() {
+        given:
+        def operationBuilder = createCustomOperationBuilder(Observable.error(new RuntimeException()))
+
+        when:
+        objectUnderTest.queue(operationBuilder).subscribe(testSubscriber)
+
+        then:
+        testSubscriber.assertError(RuntimeException.class)
```

Comment:
Yeah, but it was already covered by another test case so I decided not to repeat. Will merge the two tests together I think.